### PR TITLE
Test arch-travis on yay

### PR DESCRIPTION
**Pull request by Jguer**
Simple test of arch-travis on yay. Should be possible to start adding tests using this as a base.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+language: generic
+services:
+- docker
+
+arch:
+  packages:
+  - git
+  - go
+  script:
+  - make
```
Comment (Morganamilo) [inline .travis.yml]: No need to run `make` first, `make test` should be fine by itself.

Comment (Jguer) [reply .travis.yml]: Running `make` to test the building of `yay` which also leaves a `yay` executable ready for future system wide tests, as for what I checked, `make test` does not produce an executable

Comment (Morganamilo) [reply .travis.yml]: `go test` will fail if the binary can't build, didn't think about leaving an executable behind though.

**Comment by Morganamilo** [comment]
Requiring internet for testing is something I would really like to avoid. The state of the repos and AUR are always changing and I would rather not risk running random pkgbuilds.

As I brought up before what I would like to do is run a local AUR instance and either a frozen copy of the current arch repos or out one via `repo-add`.

I actually have a pkgbuild that I wrote for testing a local AUR instance:
```sh
#Maintainer Anna <morganamilo@gmail.com>
_randomdata=n
_pkgname=aurweb
pkgname=aurweb-test
pkgver=4.6.0
pkgrel=1
pkgdesc="Arch User Repository (AUR) Databse for predictable testing"
arch=(x86_64)
url="http://aur.archlinux.org"
license=('GPL')
depends=(php php-sqlite sqlite)
makedepends=(words fortune-mod)
source=("$pkgname-$pkgver::https://github.com/lfos/aurweb/archive/v$pkgver.tar.gz"
	"config"
	"aurweb-test"
	"php.ini"
	"aurweb.sqlite3"
	"aurweb-test.service")
md5sums=('d1551ca1118506bf80d70dd2ab7939c0'
         'cf2ddbb24d78073490f7861850432fed'
         'bfbc35d1b96d37cb42aa8417cd0647ae'
         'aa97d8f335a00858485e342768942c83'
         '495e756c43084d3445b3f8cebca1b01c'
         '9dc21501433090b455197790d1293f01')

build() {
	cd "$srcdir/$_pkgname-$pkgver/schema"

	if [ $_randomdata == "y" ]; then
		make
		./gendummydata.py out.sql
		sqlite3 ../aurweb.sqlite3 < aur-schema-sqlite.sql
		sqlite3 ../aurweb.sqlite3 < out.sql
	else
		cp "$srcdir/aurweb.sqlite3" "../aurweb.sqlite3"
	fi
}

package() {
	cd "$srcdir/$_pkgname-$pkgver"

	install -dm755 "$pkgdir/srv/$pkgname"

	for dir in aurweb/ conf/ doc/ po/ schema/ test/ upgrading/ web/; do
		cp -a "$srcdir/$_pkgname-$pkgver/$dir" "$pkgdir/srv/$pkgname"
		chmod 755 "$pkgdir/srv/$pkgname/$dir"
	done

	install -Dm644 "$srcdir/$_pkgname-$pkgver/aurweb.sqlite3" "$pkgdir/srv/$pkgname/aurweb.sqlite3"
	install -Dm644 "$srcdir/config" "$pkgdir/srv/$pkgname/conf/config"
	install -Dm755 "$srcdir/aurweb-test" "$pkgdir/srv/$pkgname/aurweb-test"
	install -Dm755 "$srcdir/php.ini" "$pkgdir/srv/$pkgname/php.ini"

	install -d "$pkgdir/usr/lib/systemd/system/"
	install -m644 "$srcdir/aurweb-test.service" "$pkgdir/usr/lib/systemd/system/"

	install -dm755 "$pkgdir/usr/bin"
	ln -s "/srv/$pkgname/aurweb-test" "$pkgdir/usr/bin/"
}
```

This doesn't cover storing pkgbuilds, just the meta data. The workaround is to have the pkgbuilds already in cache so Yay does not try to download them from the AUR instance.

**Comment by Jguer** [comment]
I just wanted to test the basic functionality to see if it was working. I'm satisfied with this. 
I'm open to suggestions on the follow up

**Comment by Morganamilo** [comment]
I'ts not run during make test so I don't mind too much. I would like the above point set up eventually though. I gets it's a lot of effort though, hence why I haven't come up with a fully working example.

**Comment by Jguer** [comment]
@Morganamilo 
```
    config_test.go:21: error: config file /home/morganamilo/git/yay/testdata/pacman.conf could not be read: No such file or directory
```

**Comment by Morganamilo** [comment]
Hmm may have put in an absolute path at first and forgot to change it.

**Pull request merged**